### PR TITLE
gtestdependency: find libraries using the compiler

### PR DESCRIPTION
**Pull request by larperaxis**
Use the compiler object to find gtest libraries. Fixes the following
cases:
- cross compiling looked in host library paths
- static libgtest was not supported

### CODE REVIEW

File: mesonbuild/dependencies/dev.py [hunk 1]
```diff
@@ -28,32 +28,39 @@
...
+        gtest_detect = self.cpp_compiler.find_library("gtest", self.env, [])
+        gtest_main_detect = self.cpp_compiler.find_library("gtest_main", self.env, [])
+        if gtest_detect and gtest_main_detect:
             self.is_found = True
             self.compile_args = []
-            self.link_args = ['-lgtest']
+            self.link_args = gtest_detect
             if self.main:
-                self.link_args.append('-lgtest_main')
+                self.link_args.append(*gtest_main_detect)
```
Comment (nirbheek) [inline mesonbuild/dependencies/dev.py]: `self.link_args += gtest_main_detect` ?

Comment (larperaxis) [inline mesonbuild/dependencies/dev.py]: Agreed, fixed and pushed.

**Comment by jpakkane** [comment]
Looks good, but please apply this change to GMock too. It is almost identical.

**Comment by larperaxis** [comment]
I made the same change to GMockDependency now and re-factored some common code to a new function.

**Comment by jpakkane** [comment]
Great, thanks.

**Pull request merged**